tests/test_crawl.py

```diff
@@ -91,12 +91,11 @@ def test_retry_conn_failed(self):
 
     @defer.inlineCallbacks
     def test_retry_dns_error(self):
-        with mock.patch('socket.gethostbyname',
-                        side_effect=socket.gaierror(-5, 'No address associated with hostname')):
-            crawler = self.runner.create_crawler(SimpleSpider)
-            with LogCapture() as l:
-                yield crawler.crawl("http://example.com/")
-            self._assert_retried(l)
+        crawler = self.runner.create_crawler(SimpleSpider)
+        with LogCapture() as l:
+            # try to fetch the homepage of a non-existent domain
+            yield crawler.crawl("http://dns.resolution.invalid/")
```

Comment:
A trailing `.` in a domain name - or the authority section of a URL, for that matter - is perfectly valid; it's used to indicate top-level termination rather than using an implicit search domain.  If the crawler isn't fetching those it's a bug somewhere in the HTTP stack.